src/main/java/org/craftercms/engine/util/logging/ThreadContextKeyExistsFilter.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.engine.util.logging;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.filter.AbstractFilter;
+import org.apache.logging.log4j.message.Message;
+
+import static org.apache.logging.log4j.core.Filter.ELEMENT_TYPE;
+import static org.apache.logging.log4j.core.Core.CATEGORY_NAME;
+
+/**
+ * Log4j filter that decides to log an entry based on whether an MDC key exists or not. If the key specified by
+ * {@code key} exists, and {@code acceptIfKeyExists} is true, then {@link Result#ACCEPT} is returned. If
+ * {@code acceptIfKeyExists} is false, {@link Result#DENY} is returned. If the key doesn't exist, and
+ * {@code denyIfKeyDoesNotExist} is true, then {@link Result#DENY} is returned, otherwise {@link Result#NEUTRAL} is
+ * returned.
+ *
+ * @author avasquez
+ */
+@Plugin(name = ThreadContextKeyExistsFilter.PLUGIN_NAME, category = CATEGORY_NAME, elementType = ELEMENT_TYPE)
+public class ThreadContextKeyExistsFilter extends AbstractFilter {
```

Comment:
Is this needed anymore? Doesn't Log4j2 already have filters that check if a MDC key exists?

Reply:
I couldn't find any, they only provide a filter to match keys to specific values  https://logging.apache.org/log4j/2.0/manual/filters.html#ThreadContextMapFilter